src/runtime/TemporalObject.cpp

```diff
@@ -556,6 +810,211 @@ bool TemporalPlainDateTime::ISODateTimeWithinLimits(ExecutionState& state, const
     return IS_IN_TIME_RANGE(ms);
 }
 
+std::map<std::string, int> TemporalPlainDateTime::balanceISODateTime(ExecutionState& state, const int& year, const int& month, const int& day, const int& hour, const int& minute, const int& second, const int& millisecond, const int& microsecond, const int& nanosecond)
+{
+    auto balancedTime = TemporalPlainTime::balanceTime(state, hour, minute, second, millisecond, microsecond, nanosecond);
+    auto balancedDate = TemporalPlainDate::balanceISODate(state, year, month, day + balancedTime["Days"]);
+    balancedDate.insert(balancedTime.begin(), balancedTime.end());
+    return balancedDate;
+}
+
+Value TemporalPlainDateTime::createTemporalDateTime(ExecutionState& state, const Value& isoYear, const Value& isoMonth, const Value& isoDay, const Value& hour, const Value& minute, const Value& second, const Value& millisecond, const Value& microsecond, const Value& nanosecond, const Value& calendar, Optional<Object*> newTarget)
+{
+    ASSERT(isoYear.isInteger(state) && isoMonth.isInteger(state) && isoDay.isInteger(state) && hour.isInteger(state) && minute.isInteger(state) && second.isInteger(state) && millisecond.isInteger(state) && microsecond.isInteger(state) && nanosecond.isInteger(state));
```

Comment:
How about directly defining these parameter types as integer instead of `Value`?
IMO we can remove redundant Value translations